torch/csrc/jit/codegen/cuda/lower_misaligned_vectorization.cpp

```diff
@@ -393,14 +395,26 @@ class MisalignedVectorizationModifier {
           fl->iter_domain(),
           fl->index(),
           ir_builder.zeroVal(),
-          stop,
+          loop_stop,
           ir_builder.oneVal(),
           vectorize && has_vectorize_op,
           vectorize_shift,
           fl->isUnrollRequired());
 
+      auto body = &new_loop->body();
```

Comment:
Wouldn't we lose access to the for-loop body, since it is replaced with the ITE then body?

I was thinking of this more conservative approach:
```
if (pred_stop != nullptr) {
  // Create new predicate and ITE
  // Add old for-loop expressions to ITE then body
  // Add ITE to new for-loop body
} else {
 // Add old for-loop expressions to new for-loop body
}
```

Reply:
Yes, `body` becomes a pointer to the then body, but that's the only scope we need, isn't it? Do we need to add anything to the loop body but not in the IfThenElse body?

Reply:
Not that I can think of, but it seems odd that the for-loop body scope is lost. In lower_unroll, the for-loop body is preserved when adding predicates. 

https://github.com/csarofeen/pytorch/blob/20_12_3_devel/torch/csrc/jit/codegen/cuda/lower_unroll.cpp#L119-L129

Reply:
Not completely sure what the issue is, but `body` is just a pointer, so we don't really lose the scope. It can be still accessed by `new_loop->body()`.